File: Leetcode/2116. Check if a Parentheses String Can Be Valid.cpp
class Solution {
public:
    bool canBeValid(string s, string locked) {
        int n=s.size();
        if (n%2!=0) return false;
        int o=0,f=0;
        for (int i=0;i<n;i++) {
            if (locked[i]=='1') {
                if (s[i]=='(') o++;
                else o--;
            } else {
                f++;
            }
            if (o+f<0) return false;
        }
        o=0,f=0;
        for (int i=n-1;i>=0;i--) {
            if (locked[i]=='1') {
                if (s[i]==')') o++;
                else o--;
            } else {
                f++;
            }
            if (o+f<0) return false;
        }
        return true;
    }
};

File: Leetcode/2116. Check if a Parentheses String Can Be Valid.py
class Solution:
    def canBeValid(self, s: str, locked: str) -> bool:
        o = 0
        f = 0
        n = len(s)
        if n % 2 == 1:
            return False
        for i in range(n):
            if locked[i] == '1':
                if s[i] == '(':
                    o += 1
                else:
                    o -= 1
            else:
                f += 1
            if o + f < 0:
                return False
        o = 0
        f = 0
        for i in range(n - 1, -1, -1):
            if locked[i] == '1':
                if s[i] == ')':
                    o += 1
                else:
                    o -= 1
            else:
                f += 1
            if o + f < 0:
                return False
        return True
